cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java

```diff
@@ -228,4 +230,16 @@ public void error(Throwable cause) {
     }
     return controller;
   }
+
+  @Nullable
+  private String getAdapterName(Arguments arguments) {
+    // TODO: Currently this logic is super ugly, should try and pass adapter name as its own separate arg
+    if (arguments.hasOption(ProgramOptionConstants.SCHEDULE_NAME)) {
+      String scheduleName = arguments.getOption(ProgramOptionConstants.SCHEDULE_NAME);
+      if (scheduleName.contains(".")) {
```

Comment:
yeah actually, not sure why schedule doesn't follow the rest. seems like it should.... But whatever separator is used will be a valid character so it has the same problem.